src/core/tests/pass/serialization/serialize.cpp

```diff
@@ -56,6 +56,35 @@ class SerializationTest : public ov::test::TestsCommon, public testing::WithPara
     }
 };
 
+TEST(SerializationTest, WriteInReadOnly) {
+    // Create read-only files
+    ASSERT_TRUE(createReadOnlyFile(m_out_xml_path));
+    ASSERT_TRUE(createReadOnlyFile(m_out_bin_path));
+
+    // Set up the serializer with the current paths
+    ov::pass::Serialize serializer(m_out_xml_path, m_out_bin_path);
+
+    auto m = std::make_shared<ov::Model>(ov::OutputVector{}, ov::ParameterVector{}, "");
+
+    // Expect that running the serializer on a read-only file throws an exception
+    EXPECT_THROW(serializer.run_on_model(m), ov::AssertFailure);
+
+    // Force removal of read-only attribute (if needed) and then remove the files.
+    auto forceRemove = [](const std::string& filename) {
```

Comment:
```suggestion
    auto force_remove = [](const std::string& filename) {
```
Use `snake_case` for variables.